### PR TITLE
releases: add TiDB release timeline

### PR DESCRIPTION
**Pull request by lilin90**
Signed-off-by: lilin90 <lilin@pingcap.com>

### What is changed, added or deleted? (Required)

- Add a new file `release-timeline.md` to show all the released TiDB versions in reverse chronological order
- Add the new file to TOC and index
- Fix typos

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: N/A
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: _index.md [hunk 1]
```diff
@@ -102,6 +102,7 @@ Designed for the cloud, TiDB provides flexible scalability, reliability and secu
 - [Privilege Management](/privilege-management.md)
 - [Role-Based Access Control](/role-based-access-control.md)
 - [Certificate-Based Authentication](/certificate-authentication.md)
+- [Release Timeline](/releases/release-timeline.md)
```
Comment (TomShawn) [inline _index.md:105]: Would it better to place the link under `<ColumnTitle>About TiDB</ColumnTitle>`?

Comment (lilin90) [reply _index.md:105]: Sounds good. 🤔

**Review by hfxsd — APPROVED** [review]

File: releases/release-timeline.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+| [2.0.2](/releases/release-2.0.2.md) | 2018-05-21 |
+| [2.0.1](/releases/release-2.0.1.md) | 2018-05-16 |
+| [2.0.0](/releases/release-2.0-ga.md) | 2018-04-27 |
+| [2.0.0-rc.5](/releases/release-2.0-rc.5.md) | 2018-04-17 |
+| [2.0.0-rc.4](/releases/release-2.0-rc.4.md) | 2018-03-30 |
+| [2.0.0-rc.3](/releases/release-2.0-rc.3.md) | 2018-03-23 |
+| [2.0.0-rc.1](/releases/release-2.0-rc.1.md) | 2018-03-09 |
+| [1.1.0-beta](/releases/release-1.1-beta.md) | 2018-02-24 |
+| [1.0.8](/releases/release-1.0.8.md) | 2018-02-11 |
+| [1.0.7](/releases/release-1.0.7.md) | 2018-01-22 |
```
[the remark this answers is not in the document]
Comment (lilin90) [reply releases/release-timeline.md:123]: @qiancai They are in the TOC in the [English docs](https://docs.pingcap.com/tidb/dev/release-1.0.8).

File: releases/release-timeline.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+| [1.1.0-alpha](/releases/release-1.1-alpha.md) | 2018-01-19 |
+| [1.0.6](/releases/release-1.0.6.md) | 2018-01-08 |
+| [1.0.5](/releases/release-1.0.5.md) | 2017-12-26 |
+| [1.0.4](/releases/release-1.0.4.md) | 2017-12-11 |
+| [1.0.3](/releases/release-1.0.3.md) | 2017-11-28 |
+| [1.0.2](/releases/release-1.0.2.md) | 2017-11-13 |
+| [1.0.1](/releases/release-1.0.1.md) | 2017-11-01 |
```
[the remark this answers is not in the document]
Comment (lilin90) [reply releases/release-timeline.md:130]: @qiancai They are in the TOC in the [English docs](https://docs.pingcap.com/tidb/dev/release-1.0.6).

File: releases/release-timeline.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
+---
+title: TiDB Release Timeline
+summary: Learn about the TiDB release timeline.
+---
+
+# TiDB Release Timeline
+
+This document shows all the released TiDB versions in reverse chronological order.
+
+| Version | Release Date |
```
[the remark this answers is not in the document]
Comment (lilin90) [reply releases/release-timeline.md:10]: Both versions are ok. I would like to keep the original version.

**Review by TomShawn — APPROVED** [review]
LGTM

File: releases/release-timeline.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (qiancai) [reply releases/release-timeline.md:10]: LGTM

File: releases/release-timeline.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (qiancai) [reply releases/release-timeline.md:130]: 那是中文 TOC 漏了....

File: releases/release-timeline.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+| [3.0.6](/releases/release-3.0.6.md) | 2019-11-28 |
+| [2.1.18](/releases/release-2.1.18.md) | 2019-11-04 |
+| [3.0.5](/releases/release-3.0.5.md) | 2019-10-25 |
+| [3.0.4](/releases/release-3.0.4.md) | 2019-10-08 |
+| [2.1.17](/releases/release-2.1.17.md) | 2019-09-11 |
+| [3.0.3](/releases/release-3.0.3.md) | 2019-08-29 |
+| [2.1.16](/releases/release-2.1.16.md) | 2019-08-15 |
+| [3.0.2](/releases/release-3.0.2.md) | 2019-08-07 |
+| [3.0.1](/releases/release-3.0.1.md) | 2019-07-16 |
+| [2.1.15](/releases/release-2.1.15.md) | 2019-07-16 |
```
[the remark this answers is not in the document]
Comment (lilin90) [reply releases/release-timeline.md:77]: Let's use the date in Chinese.

**Pull request merged**